pkg/cloud/services/networking/router.go

```diff
@@ -165,6 +165,27 @@ INTERFACE_LOOP:
 	return nil
 }
 
+func (s *Service) DeleteRouter(network *infrav1.Network) error {
+
+	if network.Router == nil || network.Router.ID == "" {
+		s.logger.V(4).Info("No need to delete router since no router exists.")
+		return nil
+	}
+
+	if network.Subnet == nil || network.Subnet.ID == "" {
+		s.logger.V(4).Info("No need to remove router interface since no subnet exists.")
+	} else {
+		_, err := routers.RemoveInterface(s.client, network.Router.ID, routers.RemoveInterfaceOpts{
+			SubnetID: network.Subnet.ID,
+		}).Extract()
+		if err != nil {
+			return fmt.Errorf("unable to remove router interface: %v", err)
+		}
+		s.logger.V(4).Info("Removed RouterInterface of Router", "id", network.Router.ID)
+	}
+	return routers.Delete(s.client, network.Router.ID).ExtractErr()
```

Comment:
Can we change that code so it doesn't fail when the router already has been deleted?

Problem might be that we delete the router are somehow not able to patch the openstackcluster and the next reconcile fails because the network is already gone.